patient-db/src/main/java/com/hartwig/hmftools/patientdb/clinical/readers/LimsPatientReader.java

```diff
@@ -0,0 +1,90 @@
+package com.hartwig.hmftools.patientdb.clinical.readers;
+
+import com.google.common.collect.Lists;
+import com.hartwig.hmftools.common.doid.DoidNode;
+import com.hartwig.hmftools.patientdb.clinical.datamodel.*;
+import com.hartwig.hmftools.patientdb.clinical.ecrf.formstatus.FormStatus;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+public class LimsPatientReader {
+
+    private static final String DOID_STRING_DELIMITER = ",";
+
+    @NotNull
+    private final List<DoidNode> doidNodes;
+
+    public LimsPatientReader(List<DoidNode> doidNodes) {
+        this.doidNodes = doidNodes;
+    }
+
+    @NotNull
+    public Patient read(@NotNull String patientIdentifier, @NotNull SampleData chosenSample, @NotNull List<SampleData> sequencedSamples) {
+        List<String> doids = Lists.newArrayList(chosenSample.limsTumorDoids().split(DOID_STRING_DELIMITER));
+
+        CuratedPrimaryTumor curatedPrimaryTumor = ImmutableCuratedPrimaryTumor.builder()
+                .location(chosenSample.limsTumorLocation())
+                .type(chosenSample.limsTumorType())
+                .extraDetails(chosenSample.limsTumorExtra())
+                .doidNodes(resolveDoidNodes(doidNodes, doids))
+                .isOverridden(false)
+                .build();
+
+        return new Patient(patientIdentifier,
+                toBaselineData(curatedPrimaryTumor),
+                noPreTreatmentData(),
+                sequencedSamples,
+                Lists.newArrayList(),
```

Comment:
Thanks, I will leave this as-is though, since the "old" style is used everywhere in this project, so we can better update these in a separate refactor commit one day I think.